### PR TITLE
Added Luminaire Versioning for Orientation Property Change

### LINKED ISSUE #221: Update versioning to align with Luminaire Property change
#### Description:
As per the title, the property Direction has been changed to Orientation with a type of Basis.

### PR DESCRIPTION
**Pull request by enarhi**
### NOTE: Depends on 
https://github.com/BHoM/BHoM/pull/1501

   
### Issues addressed by this PR

Closes #221 

Added method to converter to handle creating new Orientation Basis property from previous Direction vector property.

### CODE REVIEW

**Comment by enarhi** [comment]
@BHoMBot check versioning

File: BHoMUpgrader62/Converter.cs [hunk 1]
```diff
@@ -38,12 +38,95 @@ public class Converter : Base.Converter
...
         /***************************************************/
         /**** Private Methods                           ****/
         /***************************************************/
 
+        private static Dictionary<string, object> UpgradeLuminaire(Dictionary<string, object> oldVersion)
+        {
+            Dictionary<string, object> newVersion = new Dictionary<string, object>(oldVersion);
+            if (newVersion.ContainsKey("Direction"))
+            {
+                if (newVersion["Direction"] is null)
```
Comment (FraserGreenroyd) [inline BHoMUpgrader62/Converter.cs:54]:
```suggestion
                if (newVersion["Direction"] == null)
```

File: BHoMUpgrader62/Converter.cs [hunk 2]
```diff
@@ -38,12 +38,95 @@ public class Converter : Base.Converter
...
+                Dictionary<string, object> xVec = new Dictionary<string, object>() { ["X"] = 1, ["Y"] = 0, ["Z"] = 0 };
+                Dictionary<string, object> yVec = new Dictionary<string, object>() { ["X"] = 0, ["Y"] = 1, ["Z"] = 0 };
+                Dictionary<string, object> zVec = new Dictionary<string, object>() { ["X"] = 0, ["Y"] = 0, ["Z"] = 1 };
+
+                Dictionary<string, object> basis = new Dictionary<string, object>() { ["X"] = xVec, ["Y"] = yVec, ["Z"] = zVec };
+
+                Dictionary<string, object> orientation = newVersion["Direction"] as Dictionary<string, object>;
+                double dirX = (double)orientation["X"];
+                double dirY = (double)orientation["Y"];
+                double dirZ = (double)orientation["Z"];
```
Comment (FraserGreenroyd) [inline BHoMUpgrader62/Converter.cs:70]: Might be worth doing these in `try/catch` statements in case any of them can't cast/convert to a `double`

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

File: BHoMUpgrader62/Converter.cs (hunk 2 above)
Comment (enarhi) [reply BHoMUpgrader62/Converter.cs]: added

**Comment by enarhi** [comment]
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
@BHoMBot check copyright-compliance

**Review by FraserGreenroyd — APPROVED** [review]
Code reviewed only, makes sense and is what I would expect for the changes made in the oM.

**Pull request merged**